### PR TITLE
feat: port attachment carousel

### PR DESCRIPTION
**Pull request by juliodialpad**
# Port attachment carousel

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will increment the version number on release:

- [ ] Fix
- [x] Feature
- [ ] Performance Improvement
- [ ] Refactor

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-1434

## :book: Description

- Ported Attachment carousel from vue 3

## :bulb: Context

- Attachment carousel was created as Vue 3 only component.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [ ] I have added / updated unit tests.
- [ ] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script.
- [ ] I have validated components with a screen reader.
- [ ] I have validated components keyboard navigation.

If new component:

- I am exporting any new components or constants:
  - [x] from the index.js in the component directory.
  - [x] from the index.js in the root (either `packages/dialtone-vue2` or `packages/dialtone-vue3`).
- [ ] I have added the styles for the new component to the `packages/dialtone-css` package.
- [ ] I have created a page for the new component on the documentation site in `apps/dialtone-documentation`.
- [ ] I have added the new component to `common/components_list.cjs` in `packages/dialtone-vue2` and `packages/dialtone-vue3`
- [x] I have created a component story in storybook
- [ ] I have created story / stories for any relevant component variants in storybook
- [x] I have created a docs page for the component in storybook.
- [x] I have checked that changing all props/slots via the UI in storybook works as expected.

## :camera: Screenshots / GIFs

Screenshot 2024-01-04 at 2 22 49 p m

### CODE REVIEW

**Comment by braddialpad** [comment]
Some weird stuff happening in the percy diff. Not sure if it's flaky or it's actually related to this commit.

File: packages/dialtone-vue2/recipes/conversation_view/attachment_carousel/attachment_carousel.stories.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  See https://storybook.js.org/docs/vue/api/argtypes#manual-specification
+
+  To set the description of a slot, add the below comment to the line above where the slot is defined:
+  <!-- @slot example slot decorator -->
+*/
+
+const mediaList = [
+  {
+    type: 'image',
+    path: 'https://vue.dialpad.design/assets/test-078acfea.jpg',
```
Comment (braddialpad) [inline packages/dialtone-vue2/recipes/conversation_view/attachment_carousel/attachment_carousel.stories.js]: this is referencing vue.dialpad.design so the images aren't showing up.. guessing this is a problem on the vue 3 branch also.

**Review by braddialpad — COMMENTED** [review]
Looks good for the most part, I'll check again when the images are fixed.

File: packages/dialtone-vue2/recipes/conversation_view/attachment_carousel/attachment_carousel.stories.js (hunk 1 above)
Comment (juliodialpad) [reply packages/dialtone-vue2/recipes/conversation_view/attachment_carousel/attachment_carousel.stories.js]: Changed to a relative url, should work correctly now on vue 2 and vue 3

**Review by braddialpad — APPROVED** [review]
Looks good!

**Pull request merged**